src/mqtt_transport.erl

```diff
@@ -44,6 +44,8 @@ open(Type, Host, Port, Options) ->
     open(Type, Host, Port, Options, infinity).
 
 -spec open(tcp | ssl, host(), port_number(), tcp_options() | ssl_options(), timeout()) -> {ok, transport()} | {error, term()}.
+open(Protocol, Host, Port, Options, Timeout) when is_binary(Host) ->
+    open_tcp(unicode:characters_to_list(Host), Port, Options, Timeout);
```

Comment:
I think this should be
`open(Protocol, unicode:characters_to_list(Host), Port, Options, Timeout);`